config_containerpools.go

```diff
@@ -0,0 +1,54 @@
+package evergreen
+
+import (
+	"github.com/evergreen-ci/evergreen/db"
+	"github.com/pkg/errors"
+	"gopkg.in/mgo.v2/bson"
+)
+
+// ContainerPool holds settings for each container pool
+type ContainerPool struct {
+	// Distro of parent host that hosts containers
+	Distro string `bson:"distro" json:"distro" yaml:"distro"`
+	// ID of container pool
+	Id string `bson:"id" json:"id" yaml:"id"`
+	// Maximum number of containers per parent host with this container pool
+	MaxContainers int `bson:"max_containers" json:"max_containers" yaml:"max_containers"`
+}
+
+type ContainerPoolsConfig struct {
+	Pools []ContainerPool `bson:"pools" json:"pools" yaml:"pools"`
+}
+
+func (c *ContainerPoolsConfig) SectionId() string { return "container_pools" }
+
+func (c *ContainerPoolsConfig) Get() error {
+	err := db.FindOneQ(ConfigCollection, db.Query(byId(c.SectionId())), c)
+	if err != nil && err.Error() == errNotFound {
+		*c = ContainerPoolsConfig{}
+		return nil
+	}
+	return errors.Wrapf(err, "error retrieving section %s", c.SectionId())
+}
+
+func (c *ContainerPoolsConfig) Set() error {
+	_, err := db.Upsert(ConfigCollection, byId(c.SectionId()), bson.M{
+		"$set": bson.M{
+			"pools": c.Pools,
```

Comment:
the key should be a bsonutil.MustHaveTag

Reply:
Fixed.